Error serializing models during call to function `_serialize_polymorphic_fields`

Calling
```python
fhir_resource.model_dump_json(indent=2)
```
results in
```
/venv/lib/python3.11/site-packages/fhircraft/fhir/resources/base.py:282: in model_dump_json
    return super().model_dump_json(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E       pydantic_core._pydantic_core.PydanticSerializationError: Error serializing to JSON: PydanticSerializationError: Error calling function `_serialize_polymorphic_fields`: AttributeError: 'dict' object has no attribute '_enable_polymorphic_serialization'
```

--- a/fhircraft/fhir/resources/base.py
+++ b/fhircraft/fhir/resources/base.py
@@ -83,7 +83,7 @@ class FHIRBaseModel(BaseModel, FHIRPathMixin):
     def _serialize_polymorphic_fields(self, serializer, info) -> Any:
         """Apply polymorphic serialization to FHIR fields during serialization."""
         # Check if polymorphic serialization is enabled
-        if not self._enable_polymorphic_serialization:
+        if not isinstance(self, FHIRBaseModel) or not self._enable_polymorphic_serialization:
             return serializer(self)
 
         # Get the base serialization with warnings suppressed

